src/App.tsx

```diff
@@ -1,35 +1,82 @@
+import User from './pages/User'
+import Device from './pages/Device'
+import Docspage from './pages/Docspage'
+import DeviceSensorSettings from './pages/DevicesensorSettings';
+import { Box, createTheme, ThemeProvider } from '@mui/material'
+import DeviceSettings from './pages/DeviceSettings';
+import SettingsNetworking from './pages/SettingsNetworking'
+import SettingsMaintenance from './pages/SettingsMaintenance';
+import AppUI from './pages/App';
+import { useContext } from 'react'
+import { DevicesContext } from './context/devices.context'
 function App() {
-  const [count, setCount] = useState(0)
-
-  return (
-    <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+    const theme = createTheme({
+        palette: {
+            primary: {
+                main: '#292F3F'
+            },
+            secondary: {
+                main: '#F35E19'
+            },
+            background: {
+                default: '#F0F2F5',
+            },
+            info:{
+                main:'#499DFF'
+            },
+        },
+    });
+    const { token } = useContext(DevicesContext);
+    const creds = window.sessionStorage.getItem("creds");
+    return (
+        <ThemeProvider theme={theme}>
+            <Box bgcolor={'#F0F2F5'}>
+                <HashRouter>
+                    <Routes>
+                        {
+                            (token && creds) ? (
+                                <Route >
+                                    <Route element={<Layout/>}>
+                                        <Route path='/' element={<Navigate to='/dashboard' replace />}/>
+                                        <Route path='/dashboard' element={<Dashboard/>}/>
+                                        <Route path='/devices' element={<Devices/>}/>
+                                        <Route path='/devices/:id' element={<Device/>}/>
+                                        <Route path='/devices/:id/setting' element={<DeviceSettings/>}/>
+                                        <Route path='/devices/:id/sensors/:sensorId' element={<DeviceSensor/>}/>
+                                        <Route path='/devices/:id/sensors/:sensorId/setting' element={<DeviceSensorSettings/>}/>
+                                        <Route path='/devices/:id/actuators/:sensorId' element={<DeviceSensor/>}/>
```

Comment:
sensorId should be renamed actuatorId (and you should have specific files for Actuators, see my other comment)